sklearn/compose/_column_transformer.py

```diff
@@ -266,9 +266,10 @@ def __init__(
     def _transformers(self):
         """
         Internal list of transformer only containing the name and
-        transformers, dropping the columns. This is for the implementation
-        of get_params via BaseComposition._get_params which expects lists
-        of tuples of len 2.
+        transformers, dropping the columns.
+
+        DO NOT USE: This is for the implementation of get_params via
```

Comment:
Why the loud shouty "DO NOT USE"? Can we turn it into something more informative? Otherwise I feel like I didn't learn anything that I didn't already know from `_transformers` being a private attribute.

Reply:
Extended the comment.

This is not rendered in our docs, so the note is not meant for end users. This is for us and future poor maintainers who will be confused. I started by using this instead of `self._iter` to iterate through the transformers and then I realized I shouldn't.